tests/components/imap/test_init.py

```diff
@@ -674,6 +674,72 @@ async def test_message_is_truncated(
     assert len(event_data["text"]) == 3
 
 
+@pytest.mark.parametrize("imap_search", [TEST_SEARCH_RESPONSE])
+@pytest.mark.parametrize(
+    "imap_fetch", [(TEST_FETCH_RESPONSE_TEXT_PLAIN)], ids=["plain"]
+)
+@pytest.mark.parametrize("imap_has_capability", [True, False], ids=["push", "poll"])
+async def test_text_only_in_message_data(
+    hass: HomeAssistant, mock_imap_protocol: MagicMock, caplog: pytest.LogCaptureFixture
+) -> None:
+    """Test message data only includes text in event data."""
```

Comment:
Instead of 2 new test we can parametrize `event_message_data` and the results to assert